plugins/backend/backstage-plugin-argo-cd-backend/src/service/router.ts

```diff
@@ -198,12 +198,67 @@ export function createRouter({
         kubernetesNamespace: namespace,
       });
     } catch (e: any) {
+      console.log(e);
       return response.status(500).send({
         status: 500,
         message: e.message || 'Failed to create argo app',
       });
     }
   });
+  router.put('/updateArgo/:argoAppName', async (request, response) => {
+    const argoInstanceName: string = request.body.clusterName;
+    const namespace = request.body.namespace;
+    const projectName = request.body.projectName as string;
+    const appName = request.body.appName as string;
+    const labelValue = request.body.labelValue as string;
+    const sourceRepo = request.body.sourceRepo;
```

Comment:
const {sourceRepo} = request.body;
```suggestion
    const sourceRepo = request.body.sourceRepo;
```